### PR TITLE
new methods for prediction using cartesian cross spectra

### PR DESCRIPTION
**Pull request by agrouaze**
- [x] add new methods/scripts
- [x] remove deprecated methods/scripts
- [x] packaging of the lib
- [x] add entrypoint (example : `test-hs-wv-prediction --wvpath ./S1A_WV_OCN__2SSV_20210527T215942_20210527T220847_038079_047E7D_F72B.SAFE/measurement/s1a-wv2-ocn-vv-20210527t215957-20210527t220000-038079-047e7d-002.nc`)
- [x] add reference dataset (19Mo x2)
- [x] add unit test for the 2 periods (before after EAP update) `pytest`
- [x] add models .h5 (9.3Mo x2)
- [x] update requirement.txt

### CODE REVIEW

**Comment by agrouaze** [comment]
At this stage, a review is welcome.

**Comment by agrouaze** [comment]
Task list to do after discussion with Manuel and Amine (30Aug2024):

- [x] change requirement.txt format to `pip` : `requirement_pip.txt`
- [x] test inference with python==~~3.9.19~~ 3.9.15: unit test OK
- [x] test inference with scipy==1.6.0 , numpy==1.19.5 and tensorflow==2.7.0 : unit test OK (tested with `micromamba install numpy==1.19.5 xarray==2022.03 pandas==1.4.4 scipy==1.6.0 tensorflow==2.7.0`)
- [x] rename upper case in .py files
- [x] linter to apply on all the methods
- [x] refactor constants in separate .py file
- [x] information: the .h5 shipped in this pull request can stay but they won't be used for exploitation (only AUX_ML2 is used).
- [x] add a high level test on a wv ocn .nc file
- [x] update models and unit test with latest runs

Dead line 06 sept. 2024.

File: src/s1tools/sarhspredictor/config.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+hs_wv_model_before_WV2_EAP: "./models/S1_WV_oswCartSpec_hs_NN_model_05BeforeEAP.35_bestmodel_checkpoint_30_val_loss_0.258.h5"
```
Comment (mgoacolou) [inline src/s1tools/sarhspredictor/config.yaml:1]: these two models are not present in this PR, can you update with the good ones.

File: src/s1tools/sarhspredictor/test/test_predictions_from_ocn.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        "measurement",
+        "s1a-wv2-ocn-vv-20231003t122403-20231003t122406-050600-061886-006.nc",
+    )
+    actual_ds_hs_predictions = get_hs_inference_from_cartesian_xspectra(input_test_dataset)
+    print("ds_hs_predictions", actual_ds_hs_predictions)
+    # expected_predictions = data['yhat']
+    # test Hs
+    assert np.allclose(actual_ds_hs_predictions["oswTotalHs"].values, expected["hs"], atol=1e-02)
+    # test Hs STD
+    assert np.allclose(actual_ds_hs_predictions["oswTotalHsStdev"], expected["hsstd"], atol=1e-02)
```
Comment (mgoacolou) [inline src/s1tools/sarhspredictor/test/test_predictions_from_ocn.py:26]: Can you update the expected values after updating the config.yaml with the good models files

**Review by agrouaze — COMMENTED** [review]
@mgoacolou Thanks for the fix/refactoring.

File: src/s1tools/sarhspredictor/config.yaml (hunk 1 above)
Comment (agrouaze) [reply src/s1tools/sarhspredictor/config.yaml:1]: Models added with #189812e.

File: src/s1tools/sarhspredictor/test/test_predictions_from_ocn.py (hunk 2 above)
Comment (agrouaze) [reply src/s1tools/sarhspredictor/test/test_predictions_from_ocn.py:26]: Expected values were already updated. unit test are OK:
```bash
============================================================================================ test session starts =============================================================================================
platform linux -- Python 3.10.13, pytest-8.1.1, pluggy-1.4.0
rootdir: /home1/datahome/agrouaze/sources/git/sar_hs_predictor
configfile: pyproject.toml
collected 3 items                                                                                                                                                                                             

test_predictions.py ..                                                                                                                                                                                  [ 66%]
test_predictions_from_ocn.py .                                                                                                                                                                          [100%]

============================================================================================== warnings summary ===============================================================================================
src/s1tools/sarhspredictor/test/test_predictions_from_ocn.py::test_hs_predictions_from_ocn_file
  <frozen importlib._bootstrap>:241: RuntimeWarning: numpy.ndarray size changed, may indicate binary incompatibility. Expected 16 from C header, got 96 from PyObject

src/s1tools/sarhspredictor/test/test_predictions_from_ocn.py::test_hs_predictions_from_ocn_file
  /opt/conda-envs/mamba-l2awinddir-dev/lib/python3.10/site-packages/xarray/core/computation.py:822: RuntimeWarning: divide by zero encountered in log10
    result_data = func(*input_data)

src/s1tools/sarhspredictor/test/test_predictions_from_ocn.py::test_hs_predictions_from_ocn_file
  /opt/conda-envs/mamba-l2awinddir-dev/lib/python3.10/site-packages/xarray/core/computation.py:822: RuntimeWarning: invalid value encountered in sqrt
    result_data = func(*input_data)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
======================================================================================== 3 passed, 3 warnings in 7.11s ========================================================================================
```

**Pull request merged**